ng-dev/pr/takeover-pr/takeover-pr.ts

```diff
@@ -0,0 +1,72 @@
+import {dirname, join} from 'path';
+import {AuthenticatedGitClient} from '../../utils/git/authenticated-git-client.js';
+import {Log, bold, green} from '../../utils/logging.js';
+import {checkOutPullRequestLocally} from '../common/checkout-pr.js';
+import {fileURLToPath} from 'url';
+
+/** List of accounts that are supported for takeover. */
+const takeoverAccounts = ['angular-robot'];
+
+export async function takeoverPullRequest({
+  pr,
+  branch,
+}: {
+  pr: number;
+  branch?: string;
+}): Promise<void> {
+  /** An authenticated git client. */
+  const git = await AuthenticatedGitClient.get();
+  /** The branch name used for the takeover change. */
+  const branchName = branch ?? `pr-${pr}`;
+
+  // Make sure the local repository is clean.
+  if (git.hasUncommittedChanges()) {
+    Log.error(
+      ` ✘ Local working repository not clean. Please make sure there are no uncommitted changes`,
+    );
+    return;
+  }
+
+  // Verify that the expected branch name is available.
+  if (git.runGraceful(['rev-parse', '-q', '--verify', branchName]).status === 0) {
+    Log.error(` ✘ Expected branch name \`${branchName}\` already in use`);
+    return;
+  }
+
+  /** The pull request information from Github. */
+  const {data: pullRequest} = await git.github.pulls.get({pull_number: pr, ...git.remoteParams});
+
+  // Confirm that the takeover request is being done on a valid pull request.
+  if (!takeoverAccounts.includes(pullRequest.user.login)) {
+    Log.error(` ✘ Unable to takeover pull request from: ${bold(pullRequest.user.login)}`);
+    Log.error(`   Supported accounts: ${bold(takeoverAccounts.join(', '))}`);
+    return;
+  }
+
+  Log.info(`Checking out \`${pullRequest.head.label}\` locally`);
+  await checkOutPullRequestLocally(pr, {allowIfMaintainerCannotModify: true});
+
+  Log.info(`Setting local branch name based on the pull request`);
+  git.run(['checkout', '-q', '-b', branchName]);
+
+  Log.info('Updating commit messages to close preview pull request');
+  git.run([
+    'filter-branch',
+    '-f',
+    '--msg-filter',
+    `${getCommitMessageFilterScriptPath()} ${pr}`,
+    `${pullRequest.base.sha}..HEAD`,
+  ]);
+
+  Log.info(` ${green('✔')} Checked out pull request #${pr} into branch: ${branchName}`);
```

Comment:
What's the different to `yarn ng-dev pr checkout`? (except the closes?)

Reply:
Really just the automatically closing the previous one

Reply:
I wish we could simply use `ng-dev pr checkout --takeover` or something for this, but I see that those serve two different purposes. Leaving that to you

Reply:
I'm just a little worried we are expanding the scope of `ng-dev`, without any actual testing for these commands etc.

Reply:
I refactored for it to be part of `ng-dev pr checkout`